base/system/smss/sminit.c

```diff
@@ -1819,9 +1819,9 @@ SmpCreateDynamicEnvironmentVariables(VOID)
     if (NT_SUCCESS(Status))
     {
         /* To combine it into a single string */
-        swprintf((PWCHAR)PartialInfo->Data + wcslen((PWCHAR)PartialInfo->Data),
-                 L", %S",
-                 PartialInfo2->Data);
+        ((PWCHAR)(PartialInfo->Data))[PartialInfo->DataLength / sizeof(WCHAR)] = L'\0';        
+        wcsncat((PWCHAR)PartialInfo->Data, L", ", 2 * sizeof(WCHAR));
+        wcsncat((PWCHAR)PartialInfo->Data, (PWCHAR)PartialInfo2->Data, PartialInfo2->DataLength / sizeof(WCHAR));		
```

Comment:
So the alternative:
```suggestion
        swprintf(((PWCHAR)PartialInfo->Data) + PartialInfo->DataLength / sizeof(WCHAR),
                 L", %.*S",
                 PartialInfo2->Data, PartialInfo2->DataLength / sizeof(WCHAR));
```
would not work?

Reply:
I let you test and commit. I will no longer work on ReactOS PR's until further notice.

Reply:
Note that I've used small s instead of capital S in the swprintf string formatter, because according to all the comments made so far it appears that the string retrieved in the PartialInfo2 is already in unicode (and not in ANSI as was supposed in the original code, hence the truncation that appeared to happen).